moveit_planners/chomp/chomp_interface/CMakeLists.txt

```diff
@@ -1,72 +1,67 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.5)
 project(moveit_planners_chomp)
 
-if(NOT "${CMAKE_CXX_STANDARD}")
-  set(CMAKE_CXX_STANDARD 14)
-endif()
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+# Common cmake code applied to all moveit packages
+find_package(moveit_common REQUIRED)
+moveit_package()
 
-# find catkin in isolation so that CATKIN_ENABLE_TESTING is defined
-find_package(catkin REQUIRED)
-if(CATKIN_ENABLE_TESTING)
-  set(CHOMP_TEST_DEPS moveit_ros_planning_interface)
-else()
-  set(CHOMP_TEST_DEPS)
-endif()
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(moveit_core REQUIRED)
+find_package(chomp_motion_planner REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
+set(THIS_PACKAGE_INCLUDE_DEPENDS
   moveit_core
-  pluginlib
   chomp_motion_planner
-  ${CHOMP_TEST_DEPS}
-)
-
-find_package(Eigen3 REQUIRED)
-find_package(Boost REQUIRED)
-
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS roscpp moveit_core pluginlib
+  moveit_core
+  pluginlib
+  rclcpp
 )
 
-include_directories(include)
-include_directories(SYSTEM
-  ${catkin_INCLUDE_DIRS}
-  ${Boost_INCLUDE_DIRS}
-  ${EIGEN3_INCLUDE_DIRS}
+include_directories(
+  include
 )
 
-add_library(${PROJECT_NAME} src/chomp_interface.cpp src/chomp_planning_context.cpp)
-set_target_properties(${PROJECT_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
+set(MOVEIT_LIB_NAME moveit_chomp_interface)
 
-add_library(chomp_planner_plugin src/chomp_plugin.cpp)
-set_target_properties(chomp_planner_plugin PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(chomp_planner_plugin ${PROJECT_NAME} ${catkin_LIBRARIES})
+add_library(${MOVEIT_LIB_NAME} SHARED
+  src/chomp_interface.cpp
+  src/chomp_planning_context.cpp)
+set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
+ament_target_dependencies(${MOVEIT_LIB_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
-install(FILES chomp_interface_plugin_description.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+add_library(moveit_chomp_planner_plugin SHARED src/chomp_plugin.cpp)
+ament_target_dependencies(moveit_chomp_planner_plugin ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_chomp_planner_plugin
+  ${MOVEIT_LIB_NAME}
+)
 
-install(DIRECTORY include/chomp_interface/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+install(
+  TARGETS ${MOVEIT_LIB_NAME} moveit_chomp_planner_plugin
+  EXPORT export_${PROJECT_NAME}
```

Comment:
@tylerjw didn't the export variable name change?

Reply:
We decided to use `${PROJECT_NAME}Targets`

Reply:
Fixed.